### PR TITLE
Adding AssociationType as Server to NTP Server config

### PR DESCRIPTION
**Pull request by agvarghe-cisco**
Adding AssociationType as Server to NTP Server config

### CODE REVIEW

**Comment by bstoll** [comment]
Could you elaborate on why this change is needed?  The AssociationType leaf has a default value of [SERVER](https://github.com/openconfig/public/blob/00a27a5f0f3c472205ef293ab238e0530b610811/release/models/system/openconfig-system.yang#L679) so it seems like this change would be redundant.

**Comment by agvarghe-cisco** [comment]
> Could you elaborate on why this change is needed? The AssociationType leaf has a default value of [SERVER](https://github.com/openconfig/public/blob/00a27a5f0f3c472205ef293ab238e0530b610811/release/models/system/openconfig-system.yang#L679) so it seems like this change would be redundant.
@bstoll , AssociationType leaf if mandatory for ntp server config ,  can we have a deviation until it's fixed

**Comment by bstoll** [comment]
Yes, this could be done with a deviation.  I think I would prefer if this was documented such that the NOS does not populate default values in gNMI requests and not something specific to configuring NTP server.

**Comment by vosipchu** [comment]
Cisco, can you append code for deviation itself? One there we will do a quick lgtm round and i will commit. Thanks! 
Also, when will this default behavior be implemented?
thanks

**Comment by agvarghe-cisco** [comment]
> [vosipchu](/vosipchu)

@ vosipchu, Added AssociationType as deviation. Will let know when default behavior is implemented

**Comment by vosipchu** [comment]
Sash, Yan, need your view on this deviation. If anything is not clear, ping me with questions. Thanks
@sachendras @vcheny

**Comment by sachendras** [comment]
LGTM!
Lets also track the final release for delivery and ETA.

**Review by vcheny — APPROVED** [review]

**Review by vosipchu — APPROVED** [review]

File: internal/deviations/deviations.go [hunk 1]
```diff
@@ -67,6 +67,9 @@ import "flag"
 
 // Vendor deviation flags.
 var (
+	NtpAssociationType = flag.Bool("ntp_association_type_enabled", false,
```
Comment (liulk) [inline internal/deviations/deviations.go:70]: Please be advised that NTP must be capitalized according to Go initialism.

https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Deviation flags must begin with `--deviations_`.

Also, the help string should explain what the OpenConfig behavior is for a fully compliant device.

Fixing these issues in #762

**Pull request merged**